src/lib/invoke.js

```diff
@@ -9,6 +11,12 @@ module.exports = (key, service, action, data, isCLI) => {
     'X-Build-Meta-OS': `${os.type().toLowerCase()}@${os.release()}`,
   };
 
+  if (data['x-build-outputs'] !== undefined) {
```

Comment:
Instead of passing it through `data['x-build-outputs']`, why not just pass it as a param into invoke?

Reply:
Yeah I thought about that.. 

But the [last parameter](https://github.com/readmeio/api/blob/master/src/lib/invoke.js#L5) is [optional.](https://github.com/readmeio/api/blob/master/src/api.js#L78).

I could make it so `invoke` takes another param. But then I'd have to change [this line](https://github.com/readmeio/api/blob/master/src/api.js#L78) to be: 
```
return invoke(scope.key, scope.service, action, data, undefined, outputs).then((response) => {
```

Long story short: 
1. I didn't want to add a parameter and then have to pass in `undefined` there. 
2. I didn't want to change the order of the parameters coming in.

But if we really don't like this.. I'm happy to change it so that we pass in this object so that there isn't an order dependency anymore!
```
{ 
  key: '',
  service: '',
  action: '' ,
  data: '',
  isCLI : ''
}
```

Reply:
What about doing this: 

```node
invoke(key, service, action, data, opts)
```

And making ops have `isCLI`, `outputs` and whatever else we decide to add in the future? 

I'm also not against just making it an object.

Reply:
I quite like just making it an object.